include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp

```diff
@@ -237,32 +213,42 @@ namespace picongpu
              */
             Environment<simDim>::get().Filesystem().createDirectoryWithPermissions(
                 checkpointDirectory + "/" + this->foldername);
-            HBufCalorimeter hBufLeftParsCalorimeter(this->dBufLeftParsCalorimeter->size());
-            HBufCalorimeter hBufTotal(hBufLeftParsCalorimeter.size());
+            auto dataSize = this->dBufLeftParsCalorimeter->getDataSpace();
+            HBufCalorimeter hBufLeftParsCalorimeter(dataSize);
+            HBufCalorimeter hBufTotal(dataSize);
+
+            hBufLeftParsCalorimeter.copyFrom(*this->dBufLeftParsCalorimeter);
 
-            hBufLeftParsCalorimeter = *this->dBufLeftParsCalorimeter;
+            /* mpi reduce */
 
             /* mpi reduce */
-            (*this->allGPU_reduce)(hBufTotal, hBufLeftParsCalorimeter, pmacc::algorithm::functor::Add{});
-            if(!this->allGPU_reduce->root())
+            (*allGPU_reduce)(
+                pmacc::math::operation::Add(),
+                hBufTotal.getPointer(),
+                hBufLeftParsCalorimeter.getPointer(),
+                hBufTotal.getCurrentSize(),
+                mpi::reduceMethods::Reduce());
+
+            if(!this->allGPU_reduce->hasResult(mpi::reduceMethods::Reduce()))
                 return;
 
             std::stringstream filename;
             filename << checkpointDirectory << "/" << this->foldername << "/" << filenamePrefix << "_%T."
                      << filenameExtension;
             ::openPMD::Series series(filename.str(), ::openPMD::Access::CREATE);
 
+            auto dataSize64Bit = precisionCast<size_t>(dataSize);
```

Comment:
How do you ensure that `size_t` is 64 bit? (according to the variable name) 
I assume this is only true on 64bit systems (most systems today)